### PR TITLE
Hide private methods and namespaces with a CLI arg

### LINKED ISSUE #3: Add feature to ignore `__Private` stuff

### PR DESCRIPTION
**Pull request by lexidor**
closes #3

See the `@`reviewer block

### CODE REVIEW

**Comment by lexidor** [comment]
[`@reviewer`](https://github.com/hhvm/hh-apidoc/pull/39/files#diff-787609677d592b3d1768348f6efae937R12-R31)

**Comment by lexidor** [comment]
I'll need to split the `group_by` and maps, since the order pub > pro > pri is no longer being maintained.

File: src/PageSections/_Private/Globals.hack [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * We should not remove the methods from the class, so getMarkdown() won't find them.
+ * That is an even worse solution.
+ *
+ * getMarkDown() cannot take arguments, because it is an <<__Override>>
+ * and the constructor cannot take any new arguments either, because it is a <<__ConsistentConstruct>>.
+ * We also can't add a method to the class `->setHidePrivateMethods()`, it's constructor is never called directly.
+ * It is called via the <<__ConsistentConstruct>> of PageSection.
+ *
+ * I would like to discuss a better way to hand this boolean to that method.
+ */
```
Comment (jjergus) [inline src/PageSections/_Private/Globals.hack]: I'm not familiar with all the details so this solution may not be the best (or may not work at all), but you could maybe split the `InterfaceSynopsis` class into 2 classes like `FullInterfaceSynopsis` and `NonPrivateInterfaceSynopsis` (with a common base class to cover the shared parts, or one could extend the other).

From what I remember, passing down the CLI flag to wherever we build the list of PageSections shouldn't be too hard, so you could conditionally include one or the other class in the list there.

Comment (jjergus) [reply src/PageSections/_Private/Globals.hack]: Or possibly you could create a shape/class with options (right now just the one option) and pass it down to all the PageSection classes (through that consistent constructor), even though the InterfaceSynopsis would be the only one using the options right now...

Comment (lexidor) [reply src/PageSections/_Private/Globals.hack]: I'd go with the second one. That seems to be cleaner right now.
If we have something in the future that is a drastic change from the default behavior, we might want to implement alternative classes.

**Pull request merged**